brian2/input/spikegeneratorgroup.py

```diff
@@ -245,7 +262,7 @@ def _check_args(self, indices, times, period, N, sorted):
                                                         len(times)))
         if period < 0*second:
             raise ValueError('The period cannot be negative.')
-        elif len(times) and period <= np.max(times):
+        elif len(times) and period != 0*second and period <= np.max(times):
```

Comment:
This is wrong, and the example test below exactly shows why it's wrong. What we want to do, after having computed ``period_bins`` and ``timebins`` is to check that ``period_bins<=max(timebins)``.